### ISSUE TITLE
new map

### ISSUE THREAD

**Issue by hhrhhr** [MartinSGill/SubnauticaWatcher#7]
I want to suggest using a relief map of the seabed. A "flat" biome map, in my opinion, is not at all interesting.

I converted the file _...\Subnautica\AssetBundles\lod3worldmeshes_, which contains model of octets with a size of 160 m with a resolution of 4 m. After removing unnecessary and erroneous octets and merging them into one, it is possible to create a height map. Below are examples of ready-made maps.

Top view, applied filter Ambient Occlusion (can be combined with a biome map for the effect of relief):
![ao_map](https://user-images.githubusercontent.com/966048/34075844-64b48970-e2e4-11e7-8de3-3db42df10eea.jpg)

And here the coloring of each point is added depending on its height:
![relief_map](https://user-images.githubusercontent.com/966048/34075843-6117d380-e2e4-11e7-81f4-af8a3875f403.jpg)

**Comment by hhrhhr** [comment]
But since the game is extremely "vertical", one card is not enough for orienting by the height (depth). Here we can use the "side view":
![side_map](https://user-images.githubusercontent.com/966048/34076203-94c6bd30-e2ef-11e7-9ca6-64ff1ad2d4d8.jpg)

Or even something like X-ray:
![x-ray_maps](https://user-images.githubusercontent.com/966048/34076208-aca560a0-e2ef-11e7-96e1-de8d67e292bd.jpg)

If you are interested, you can suggest changing the colors in the depth gradient or specifying the desired size of the map (or tiles). I hope that this will be useful.

**Comment by MartinSGill** [comment]
I agree. The perfect map would actually be entirely 3D, but that's beyond my skill set.

I solve this issue by having multiple "layers" that switch automatically. So when you enter the ILZ or the lost river, or the shroom caves then you get a different map, specific to that zone/area.

I'm not sure the other maps would add all that much, although the X-Ray map might be interesting to add as an alternative. It looks out of data though, the east entrance to the Lost River isn't shaped like that anymore.

**Comment by hhrhhr** [comment]
Out of data? LOD3 used in Seaglide's radar and Scanner Room, if the source (AssetBundles) is no longer relevant, then on the radar in the game will be displayed incorrect data.
By the way, what do you compare with? If with the image _lost-river-polygon.png_, how and from what it was generated?

**Comment by MartinSGill** [comment]
Meant out of date.

I created the maps I use (other than the all biome one) by actually travelling around the area and recording my route, then drawing it based on those points and the biome reported for each. Not brilliant, but fairly accurate. 

I have to admit though, they look gorgeous. How did you render them, is the code available somewhere?

**Comment by hhrhhr** [comment]
> Meant out of date

Yes you are right. Some entrances to underwater caves are really closed or moved to other places. It turns out that the "source" used by me is completely irrelevant.
I'll try to process the LOD level higher. Still, with ready 3D models (meshes) it's easier to work than to reconstruct the game world from octets ;)

> How did you render them, is the code available somewhere?

1. extract all meshes from ...\Subnautica\AssetBundles\lod3worldmeshes with AssetsBundleExtractor 2.2beta1 (or any Unity model converter can be used)
2. load these models into Meshlab, remove unnecessary (bottom at the level of three kilometers, side "walls"), clean (combine vertices, fix a bunch of irregular surfaces).
3. in Meshlab can be visualized AO and beautiful X-Ray shader, but to build a depth map I used CloudCompare v2.10.alpha
Unfortunately, ready-made models of the world are too big so that I can share them somewhere.

**Comment by hhrhhr** [comment]
> How did you render them, is the code available somewhere?
1. extract all meshes from ...\Subnautica\AssetBundles\lod3worldmeshes with AssetsBundleExtractor 2.2beta1 (or any Unity model converter can be used)
2. load these models into Meshlab, remove unnecessary (bottom at the level of three kilometers, side "walls"), clean (combine vertices, fix a bunch of irregular surfaces).
3. in Meshlab can be visualized AO and beautiful X-Ray shader, but to build a depth map I used CloudCompare v2.10.alpha
Unfortunately, ready-made models of the world are too big so that I can share them somewhere.

**Comment by dumbdiscord** [comment]
Wow this is incredibly neat! You should consider joining our modding discord! We'd love to have you. https://discord.gg/9MfKs2s
I've been making something sort of related that involves extracting the octree data from the compiled batch data and generating a 3d mesh using my implementation of a dual contouring algorithm with a catmull clark subdivision pass, which I guess could then be exported like this. I'd love to talk more about the specifics on the discord.

**Comment by nesrak1** [comment]
What could be really cool if to take a framework like threejs or just plain webgl and get an 3d x-ray view of where you are.

**Comment by hhrhhr** [comment]
updated maps:

[![Album on Imgur](https://user-images.githubusercontent.com/966048/34143935-4926636e-e4a0-11e7-820c-ded4ce67466c.jpg)](https://imgur.com/a/PfpEQ)

Used lod**4**worldmeshes and now everything looks much better (except for the area under the "Aurora"). I also tried to apply an array of points from /subnautica-map/data-source/*.json, мatches wherever possible:

[![Another album on Imgur](https://user-images.githubusercontent.com/966048/34144876-9ed1e154-e4a4-11e7-9fc6-1c94fe759411.jpg)](https://imgur.com/a/eqeGj)

If somewhere to get the biome boundaries for each (or necessary) voxel/batch, then in addition to the depth, the map is also automatically colored according to the type of biome. If the developers again dig out new caves and tunnels, I'm afraid to imagine how long it takes to manually re-navigate along the borders. And this is if after an early release there will be a debug-console and the opportunity to be injected into the game code.

**Comment by theFisher86** [comment]
I've taken the existing images and overlayed them with the ones provided by @hhrhhr .  I've also added labels to the maps as I felt that was very needed.  You can see them [here](https://imgur.com/a/Jz2Cz).  @hhrhhr didn't upload a topography of the other layers but if he can I'd be happy to add them in.

**Comment by newman55** [comment]
Hey. Can I use your maps in my mod? 
https://www.nexusmods.com/subnautica/mods/12

**Comment by theFisher86** [comment]
@newman55 fine with me but I used @hhrhhr topology stuff to make it so I'd recommend getting his approval as well.

**Comment by TheDutchDev** [comment]
Hi, I was wondering if anyone knows where I can find the lod3worldmeshes file, it doesn't seem to be located in the AssetBundles folder anymore and when I search for the file windows returns 0 results, did the subnautica devs remove the file or something? if so is there a different file I can use or could someone send me the lod3worldmeshes file?

@hhrhhr 

Thanks,

Niels

**Comment by Fonserbc** [comment]
It looks that the world files were hidden in recent updates. I can't seem to find it anywhere in the assetbundles or streaming assets.

I really wanted to make a 3d visualization map for myself :(

Did you end up finding it @sampniels ?

**Comment by newman55** [comment]
@Fonserbc 
Yes, these files are removed from the game. But what @hhrhhr did still remained. See the link.
https://forums.unknownworlds.com/discussion/147269/maps-in-games-folder/p2
